pages/index.tsx

```diff
@@ -1,31 +1,123 @@
+import { useState, useEffect, useMemo, useCallback } from 'react';
+import Modal from 'react-modal';
 import styles from '../styles/Home.module.scss';
-import { home_page } from '../config';
+import { home_page, website_config } from '../config';
 // Internal Components
+import Image from '../components/Image';
 import NavBar from '../components/NavBar';
 import EventBanner from '../components/EventBanner';
 import HeroBanner from '../components/HeroBanner';
 import Footer from '../components/Footer';
 import Section from '../layouts/Section';
+import Link from 'next/link';
+
+Modal.setAppElement('#__next'); // Set the app element for accessibility
 
 export default function Home() {
   const { sections } = home_page;
+  const [modalIsOpen, setModalIsOpen] = useState(false);
```

Comment:
can we invert this logic, to prevent paint flash. This is overly complicated but really simple but to prevent flash I think we should do something like this:
```
enum ModalState {
  Unknown,
  Open,
  Closed
}
```
and then it defaults to `Unknown` which is just the blurred backdrop with no modal. open is open and closed is obviously closed.
This just prevents the quick paint flash is all, but i dont want to always show the modal.

Alterantively can you maybe give what I suggest in the comment below a try.